File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/resource/KotlinCommonResources.java
package ru.pocketbyte.locolaser.platform.kotlinmobile.resource;

import ru.pocketbyte.locolaser.platform.kotlinmobile.resource.file.KotlinCommonResourceFile;
import ru.pocketbyte.locolaser.resource.file.ResourceFile;

import java.io.File;
import java.util.Set;

public class KotlinCommonResources extends AbsKotlinPlatformResources {

    public KotlinCommonResources(File dir, String name) {
        super(dir, name);
    }

    @Override
    protected ResourceFile[] getResourceFiles(Set<String> locales) {
        return new ResourceFile[] {
                new KotlinCommonResourceFile(getFile(), this.className, this.classPackage)
        };
    }
}

File: platform-kotlin-mpp/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmpp/resource/KotlinCommonResources.kt
package ru.pocketbyte.locolaser.platform.kotlinmpp.resource

import ru.pocketbyte.locolaser.platform.kotlinmpp.resource.file.KotlinCommonResourceFile
import ru.pocketbyte.locolaser.resource.file.ResourceFile
import ru.pocketbyte.locolaser.resource.formatting.FormattingType
import ru.pocketbyte.locolaser.resource.formatting.JavaFormattingType

import java.io.File

class KotlinCommonResources(
        dir: File,
        name: String,
        filter: ((key: String) -> Boolean)?
) : AbsKotlinPlatformResources(dir, name, filter) {

    override val formattingType: FormattingType = JavaFormattingType

    override fun getResourceFiles(locales: Set<String>?): Array<ResourceFile>? {
        return arrayOf(KotlinCommonResourceFile(
                directory, this.className, this.classPackage, formattingType))
    }
}
